### ISSUE TITLE
Unify naming of unit test files

### ISSUE THREAD

**Issue by gavinevans** [metoppv/improver#506]
As an IMPROVER developer I want to have consistently named unit test files (following a clear convention) so that developers know the standard that they are trying to adhere to, and are able to quickly find unit tests relating to a particular part of the code.

Acceptance criteria:
 * Update unit tests to adhere to the structure described in https://github.com/metoppv/improver/wiki/Code-Style-Guide#unit-test-structure.
 * Clarify the description in https://github.com/metoppv/improver/wiki/Code-Style-Guide#unit-test-structure in case of ambiguities and provide clear examples where possible.

Related issues: #467

Description:
Unit test structure is described in https://github.com/metoppv/improver/wiki/Code-Style-Guide#unit-test-structure.

Common inconsistencies in the code are:
 * Use of snake_case (lowercase with underscores) for tests on classes.
 * Unit tests for plugins in files named by the file e.g. MyPlugin in my_file.py tested in a file named test_my_file.py, rather than test_MyPlugin.py.
 * Inconsistent directory structure, primarily, inconsistent use of a sub-directories e.g. tests/utilities/solar/test_DayNightMask.py or tests/utilities/test_DayNightMask.py

In summary:
For plugins (classes), each plugin should have its own unit tests with the name of the file matching the case of the class e.g. test_DayNightMask.py. 
For functions outside classes, each function should have its unit tests within a file matching the name of the file containing the function. For example, if the function _slice_over_coordinate is in cube_manipulation.py then unit tests for would be in a file named test_cube_manipulation.py.
#467 introduced a slight change to this structure for the case where a file contains a mix of plugins and functions by introducing a sub-directory, such that a file called solar.py with a DayNightMask class and other functions contains tests in tests/utilities/solar/test_DayNightMask.py and tests/utilities/solar/test_solar.py.

**Comment by fionaRust** [comment]
Closing as no longer important.
